src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/service/GeneralOrderService.java

```diff
@@ -329,35 +354,53 @@ public boolean isOrderSharedWithIntervener4(FinremCaseDetails caseDetails) {
+     * applies regardless of this assumption.</p>
+     *
+     * @param caseDetails    the case details containing the case data
+     * @param selectedOrders the list of orders selected to be shared
+     * @return a pair containing two lists of case documents:
+     *         - the first list contains legacy orders,
+     *         - the second list contains new orders
+     */
+    public Pair<List<CaseDocument>, List<CaseDocument>> hearingOrdersToShare(FinremCaseDetails caseDetails, List<OrderToShare> selectedOrders) {
         FinremCaseData caseData = caseDetails.getData();
         final List<CaseDocument> legacyOrders = new ArrayList<>();
         final List<CaseDocument> newOrders = new ArrayList<>();
 
         List<DirectionOrderCollection> hearingOrders = caseData.getUploadHearingOrder();
-        List<FinalisedOrderCollection> finalisedOrders = caseData.getDraftOrdersWrapper().getFinalisedOrdersCollection();
+        List<FinalisedOrderCollection> finalisedOrders = ofNullable(caseData.getDraftOrdersWrapper().getFinalisedOrdersCollection())
+            .orElse(List.of());
         List<AgreedDraftOrderCollection> agreedDraftOrderCollection = emptyIfNull(caseData.getDraftOrdersWrapper().getAgreedDraftOrderCollection())
             .stream().filter(a -> a.getValue().getOrderStatus() == PROCESSED).toList();
 
-        if (selectedDocs != null) {
-            List<DynamicMultiSelectListElement> selected = selectedDocs.getValue();
-            selected.forEach(checkboxSelected -> {
-                boolean addedToOrders = populateSelectedUploadHearingOrder(legacyOrders, hearingOrders, checkboxSelected);
-                if (!addedToOrders) { // if addedToOrders equals to true then break the loop
-                    addedToOrders = populateSelectedFinalisedOrders(newOrders, finalisedOrders, checkboxSelected);
+        if (selectedOrders != null) {
+            if (selectedOrders.stream().anyMatch(s -> YesOrNo.isYes(s.getDocumentToShare()))) {
+                log.warn("It assumes that the provided selectedOrders should have a value of 'Yes' in documentToShare;"
+                    + " however, this logic filters them regardless.");
+            }
+            selectedOrders.stream().filter(o -> YesOrNo.isYes(o.getDocumentToShare())).forEach(selected -> {
+                boolean isProcessed = populateSelectedUploadHearingOrder(legacyOrders, hearingOrders, selected);
+                if (!isProcessed) {
+                    isProcessed = populateSelectedFinalisedOrders(newOrders, finalisedOrders, selected);
                 }
-                if (!addedToOrders) {
-                    populateSelectedProcessedOrders(newOrders, agreedDraftOrderCollection, checkboxSelected);
+                if (!isProcessed) {
+                    populateSelectedProcessedOrders(newOrders, agreedDraftOrderCollection, selected);
                 }
             });
         }
         return Pair.of(legacyOrders, newOrders);
     }
```

Comment:
Bit confused by the warn. If you are assuming that selectedOrders should have a value of 'Yes', then it seems we will be logging this a lot.  Perhaps replace with a comment?

Reply:
I think it should be `selectedOrders.stream().anyMatch(s -> !YesOrNo.isYes(s.getDocumentToShare()))`.